File: Week 2/Readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int compute_grade(string text);

int main(void)
{
    // Get input text from user
    string text1 = get_string("Text: ");

    // grading the test
    int grade = compute_grade(text1);

    // Print the grade of the text
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
     else if (grade >= 16)
     {
         printf("Grade 16+\n");
}
     else {
         printf("%s%i\n", "Grade ", grade);
     }

}


int compute_grade(string text)
{
    // calculating the grade
    int numLetters = 0;
    int numWords = 1;
    int numSentences = 0;


    for (int i = 0; i < strlen(text); i++)
    {

     if (isupper(text[i]) || islower(text[i]))
     {

      numLetters += 1;

     }

      else if (text[i] == 32)
      {

       numWords += 1;

  }

     else if (text[i] == 33 || text[i] == 46 || text[i] == 63)
     {

       numSentences += 1;

     }

    }

    float L = ((float) numLetters / numWords) * 100;
    float S = ((float) numSentences / numWords) * 100;

  float beforeRound = 0.0588 * L - 0.296 * S - 15.8;
  int afterRound = round(beforeRound);

    return afterRound;
    
}

File: Week 6/Problem set 6/readability.py
from cs50 import get_string  


def main():
    text1 = get_string("Text: ")
    
    grade = compute_grade(text1)
    
    if grade < 1:
        print("Before Grade 1")
    
    elif grade >= 16:
        print("Grade 16+")
        
    else:
        print(f"Grade, {grade}")    


def compute_grade(text):
    # calculating the grade
    numLetters = 0
    numWords = 1
    numSentences = 0

    for i in range(0, len(text), 1):
    
        if (text[i].isupper() or text[i].islower()):
            numLetters += 1

        elif (text[i] == " "):
            numWords += 1

        elif (text[i] == "!" or text[i] == "." or text[i] == "?"):
    
            numSentences += 1

    L = (numLetters / numWords) * 100
    S = (numSentences / numWords) * 100

    beforeRound = 0.0588 * L - 0.296 * S - 15.8
    afterRound = round(beforeRound)

    return afterRound
    
    
main()